drivers/usb/device/usb_dc_mcux_ehci.c

```diff
@@ -481,7 +479,7 @@ int usb_dc_ep_read_wait(uint8_t ep, uint8_t *data, uint32_t max_data_len,
 		return -EBUSY;
 	}
 
-	if ((ep_idx >= NUM_OF_EP_MAX) || (EP_ADDR2DIR(ep) != USB_EP_DIR_OUT)) {
+	if ((ep_idx >= NUM_OF_EP_MAX) || (USB_EP_GET_DIR(ep) != USB_EP_DIR_OUT)) {
```

Comment:
What about?
``` diff
- 	if ((ep_idx >= NUM_OF_EP_MAX) || (USB_EP_GET_DIR(ep) != USB_EP_DIR_OUT)) {
+	if ((ep_idx >= NUM_OF_EP_MAX) || (USB_EP_DIR_IS_IN(ep))) {
```
There are only 2 directions so this makes sense and is more readable.

Reply:
No, I will not change this logic here. Also not relevant to this PR.